### PR TITLE
[SPARK-41876][CONNECT][PYTHON] `test_dataframe` should catch both `AnalysisException` and `SparkConnectAnalysisException`

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
When using `test_dataframe.py` to test spark connect's python API, an error occurs.
```
Traceback (most recent call last):
  File "/Users/s.singh/personal/spark-oss/python/pyspark/sql/tests/test_dataframe.py", line 1486, in test_to
    self.assertRaisesRegex(
AssertionError: AnalysisException not raised by <lambda> 
```

The root reason is Spark connect covert `AnalysisException` to `SparkConnectAnalysisException`.
This PR lets `test_dataframe` catch both `AnalysisException` and `SparkConnectAnalysisException`.

### Why are the changes needed?
Lets `test_dataframe` catch both `AnalysisException` and `SparkConnectAnalysisException`.

### Does this PR introduce _any_ user-facing change?
'No'.
New feature.

### How was this patch tested?
Manual tests.

### CODE REVIEW

File: python/pyspark/sql/tests/connect/test_parity_dataframe.py [hunk 1]
```diff
@@ -120,8 +120,7 @@ def test_same_semantics_error(self):
     def test_sample(self):
         super().test_sample()
 
-    # TODO(SPARK-41875): throw proper errors in Dataset.to()
```
Comment (itholic) [inline python/pyspark/sql/tests/connect/test_parity_dataframe.py:123]: So, it sounds like now we throws proper errors, but the test still disabled ??

If so, maybe we should say like:
```python
# TODO(SPARK-XXXXX): Implement Dataset `to`
@unittest.skip("Fails in Spark Connect, should enable.")
```

??

Comment (HyukjinKwon) [reply python/pyspark/sql/tests/connect/test_parity_dataframe.py:123]: Yeah we should fix the test like https://github.com/apache/spark/pull/39450#discussion_r1064065514

Comment (HyukjinKwon) [reply python/pyspark/sql/tests/connect/test_parity_dataframe.py:123]: I meant that exception. We should ideally fix the test to catch both `AnalysisException` and `SparkConnectAnalysisException`. So let's file a new JIRA, and fix the comment here

File: python/pyspark/sql/tests/test_dataframe.py [hunk 2]
```diff
@@ -42,6 +42,7 @@
     DayTimeIntervalType,
 )
 from pyspark.sql.utils import AnalysisException, IllegalArgumentException
+from pyspark.sql.connect.client import SparkConnectAnalysisException
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_dataframe.py:45]: The problem here is that PyArrow is an optional dependency but importing this requires PyArrow. So we should wait for https://github.com/apache/spark/pull/39387 to be merged first, and the exception is factored out to the separate module that doesn't require PyArrow to be imported.

Comment (beliefer) [reply python/pyspark/sql/tests/test_dataframe.py:45]: I see.

Comment (HyukjinKwon) [reply python/pyspark/sql/tests/test_dataframe.py:45]: https://github.com/apache/spark/pull/39387 is merged. Could we move the Spark Connect errors into there, see fix the test cases here? See also https://github.com/apache/spark/pull/39591. cc @itholic

Comment (beliefer) [reply python/pyspark/sql/tests/test_dataframe.py:45]: OK

**Comment by beliefer** [comment]
@HyukjinKwon https://github.com/apache/spark/pull/39693 fixed the issue.

**Pull request closed without merging**